api/src/test/java/org/openmrs/api/VisitServicePerformanceTest.java

```diff
@@ -0,0 +1,102 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.api;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.openmrs.Patient;
+import org.openmrs.Visit;
+import org.openmrs.VisitType;
+import org.openmrs.api.context.Context;
+import org.openmrs.parameter.VisitSearchCriteria;
+import org.openmrs.test.jupiter.BaseContextSensitiveTest;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * Performance test demonstrating N+1 query problem resolution in VisitService.
+ * This test validates the optimization achieved through eager fetching of encounters.
+ *
+ * @since 2.7.0
+ */
+public class VisitServicePerformanceTest extends BaseContextSensitiveTest {
+
+    @Autowired
+    private VisitService visitService;
+
+    @Autowired
+    private PatientService patientService;
+
+    @BeforeAll
+    public static void enableSqlLogging() {
+        System.setProperty("hibernate.show_sql", "true");
+    }
+
+    /**
+     * Tests the performance improvement when fetching visits with eager loading of encounters.
+     * This test demonstrates the resolution of the N+1 query problem by using LEFT JOIN FETCH.
+     */
+    @Test
+    public void shouldDemonstrateSlowPerformanceWithManyVisits() {
```

Comment:
This doesn't seem like a very useful test to demonstrate that this fixes the N+1 query problem. The problem doesn't appear because there are a large number of visits, but because the visits might have a large number of encounters (or other downstream objects). However, none of the visits here have any encounters.